feat:(docs) Adds pattern doc on Basic Recomposition (#31861)
microsoft/fluentui: docs/react-v9/contributing/patterns/basic-recomposition.md

> ## Basic Recomposition

Reviewer: ## [🕵 fluentui-web-components-v3 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31861&commitId=4ecb798a0d12e7092808dbd7f932d29a50a5993c&env=prod&clientType=FLUENTUI)

  <details><summary><b>Accordion</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Accordion.Size - Dark Mode.normal.chromium.png|2| Changed|
|Accordion.Size - RTL.normal.chromium.png|2| Changed|
|Accordion.Size.normal.chromium.png|2| Changed|

</details>

  <details><summary><b>Avatar</b> 8 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar.Color.normal.chromium.png|1| Changed|
|Avatar.Colorful - Dark Mode.normal.chromium.png|452| Changed|
|Avatar.Active Appearance - Dark Mode.normal.chromium.png|2| Changed|
|Avatar.Colorful - RTL.normal.chromium.png|326| Changed|
|Avatar.Active.normal.chromium.png|5| Changed|
|Avatar.Colorful.normal.chromium.png|325| Changed|
|Avatar.Color - RTL.normal.chromium.png|1| Changed|
|Avatar.Active Appearance.normal.chromium.png|10| Changed|

</details>

  <details><summary><b>Badge</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Badge.Appearance - Dark Mode.normal.chromium.png|15| Changed|
|Badge.Appearance.normal.chromium.png|1| Changed|
|Badge.Shape.normal.chromium.png|1| Changed|
|Badge.Default.normal.chromium.png|1| Changed|
|Badge.Size.normal.chromium.png|6| Changed|
|Badge.Color.normal.chromium.png|1| Changed|

</details>

  <details><summary><b>Button</b> 75 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Button.Large Icon Only.default.chromium.png|5| Changed|
|Button.Large Icon Only.hover.chromium.png|4| Changed|
|Button.Large With Icon - Dark Mode.hover.chromium.png|731| Changed|
|Button.Large With Icon - RTL.default.chromium.png|795| Changed|
|Button.Large Icon Only - Dark Mode.hover.chromium.png|3| Changed|
|Button.Large Icon Only - Dark Mode.pressed.chromium.png|5| Changed|
|Button.Large Icon Only - Dark Mode.default.chromium.png|4| Changed|
|Button.Large With Icon.pressed.chromium.png|707| Changed|
|Button.Large Icon Only.pressed.chromium.png|3| Changed|
|Button.Medium With Icon - Dark Mode.hover.chromium.png|720| Changed|
|Button.Medium With Icon - Dark Mode.pressed.chromium.png|813| Changed|
|Button.Large With Icon - RTL.hover.chromium.png|761| Changed|
|Button.Large With Icon - Dark Mode.pressed.chromium.png|822| Changed|
|Button.Large With Icon.default.chromium.png|795| Changed|
|Button.Large With Icon.hover.chromium.png|761| Changed|
|Button.Large With Icon - RTL.pressed.chromium.png|707| Changed|
|Button.Medium With Icon - Dark Mode.default.chromium.png|775| Changed|
|Button.Medium With Icon - RTL.default.chromium.png|788| Changed|
|Button.Outline Disabled - Dark Mode.default.chromium.png|159| Changed|
|Button.Outline Disabled Focusable - Dark Mode.pressed.chromium.png|159| Changed|
|Button.Medium With Icon.hover.chromium.png|753| Changed|
|Button.Medium With Icon - RTL.hover.chromium.png|753| Changed|
|Button.Medium With Icon - RTL.pressed.chromium.png|697| Changed|
|Button.Outline Disabled - Dark Mode.hover.chromium.png|159| Changed|
|Button.Large With Icon - Dark Mode.default.chromium.png|782| Changed|
|Button.Outline Disabled Focusable - Dark Mode.hover.chromium.png|159| Changed|
|Button.Primary Disabled - Dark Mode.default.chromium.png|44| Changed|
|Button.Primary Disabled - Dark Mode.hover.chromium.png|44| Changed|
|Button.Medium With Icon.default.chromium.png|788| Changed|
|Button.Primary Disabled Focusable - Dark Mode.hover.chromium.png|44| Changed|
|Button.Primary Disabled Focusable - Dark Mode.default.chromium.png|44| Changed|
|Button.Primary Disabled - Dark Mode.pressed.chromium.png|44| Changed|
|Button.Primary Disabled Focusable.pressed.chromium.png|15| Changed|
|Button.Primary Disabled Focusable.hover.chromium.png|15| Changed|
|Button.Primary Disabled Focusable - Dark Mode.pressed.chromium.png|44| Changed|
|Button.Outline Disabled Focusable - Dark Mode.default.chromium.png|159| Changed|
|Button.Primary Disabled.pressed.chromium.png|15| Changed|
|Button.Medium With Icon.pressed.chromium.png|697| Changed|
|Button.Outline Disabled - Dark Mode.pressed.chromium.png|159| Changed|
|Button.Primary Disabled.hover.chromium.png|15| Changed|
|Button.Small With Icon - Dark Mode.pressed.chromium.png|351| Changed|
|Button.Small With Icon.pressed.chromium.png|301| Changed|
|Button.Small With Icon - Dark Mode.hover.chromium.png|311| Changed|
|Button.Small With Icon.hover.chromium.png|324| Changed|
|Button.Small With Icon - Dark Mode.default.chromium.png|334| Changed|
|Button.Subtle Disabled Focusable - Dark Mode.default.chromium.png|187| Changed|
|Button.Primary Disabled Focusable.default.chromium.png|15| Changed|
|Button.Small With Icon - RTL.hover.chromium.png|324| Changed|
|Button.Subtle Disabled Focusable - Dark Mode.pressed.chromium.png|187| Changed|
|Button.Subtle Disabled Focusable.pressed.chromium.png|25| Changed|
|Button.Subtle Disabled.default.chromium.png|25| Changed|
|Button.Subtle Disabled - Dark Mode.default.chromium.png|187| Changed|
|Button.Subtle Disabled Focusable - Dark Mode.hover.chromium.png|187| Changed|
|Button.Subtle Disabled Focusable.hover.chromium.png|25| Changed|
|Button.Transparent Disabled - Dark Mode.default.chromium.png|199| Changed|
|Button.Subtle Disabled - Dark Mode.pressed.chromium.png|187| Changed|
|Button.Small With Icon - RTL.default.chromium.png|339| Changed|
|Button.Subtle Disabled Focusable.default.chromium.png|25| Changed|
|Button.Small With Icon.default.chromium.png|339| Changed|
|Button.Transparent Disabled Focusable - Dark Mode.hover.chromium.png|199| Changed|
|Button.Transparent Disabled.default.chromium.png|27| Changed|
|Button.Transparent Disabled - Dark Mode.hover.chromium.png|199| Changed|
|Button.Transparent Disabled.hover.chromium.png|27| Changed|
|Button.Subtle Disabled.pressed.chromium.png|25| Changed|
|Button.Primary Disabled.default.chromium.png|15| Changed|
|Button.Subtle Disabled - Dark Mode.hover.chromium.png|187| Changed|
|Button.Transparent Disabled Focusable.pressed.chromium.png|27| Changed|
|Button.Small With Icon - RTL.pressed.chromium.png|301| Changed|
|Button.Transparent Disabled Focusable.hover.chromium.png|27| Changed|
|Button.Subtle Disabled.hover.chromium.png|25| Changed|
|Button.Transparent Disabled Focusable.default.chromium.png|27| Changed|
|Button.Transparent Disabled Focusable - Dark Mode.pressed.chromium.png|199| Changed|
|Button.Transparent Disabled Focusable - Dark Mode.default.chromium.png|199| Changed|
|Button.Transparent Disabled.pressed.chromium.png|27| Changed|
|Button.Transparent Disabled - Dark Mode.pressed.chromium.png|199| Changed|

</details>

  <details><summary><b>Checkbox</b> 26 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Checkbox.Checked - Dark Mode.normal.chromium.png|982| Changed|
|Checkbox.Checked.normal.chromium.png|4855| Changed|
|Checkbox.Circular Disabled Checked.normal.chromium.png|4728| Changed|
|Checkbox.Circular - Dark Mode.normal.chromium.png|792| Changed|
|Checkbox.Circular Indeterminate - Dark Mode.normal.chromium.png|936| Changed|
|Checkbox.Circular Indeterminate - Dark Mode.indeterminate.chromium.png|1102| Changed|
|Checkbox.Circular Indeterminate.indeterminate.chromium.png|4869| Changed|
|Checkbox.Circular Checked.normal.chromium.png|4883| Changed|
|Checkbox.Circular Indeterminate.normal.chromium.png|4786| Changed|
|Checkbox.Circular.normal.chromium.png|4734| Changed|
|Checkbox.Circular - RTL.normal.chromium.png|4733| Changed|
|Checkbox.Disabled Checked - Dark Mode.normal.chromium.png|777| Changed|
|Checkbox.Default.normal.chromium.png|4751| Changed|
|Checkbox.Circular Checked - Dark Mode.normal.chromium.png|1080| Changed|
|Checkbox.Indeterminate - Dark Mode.indeterminate.chromium.png|993| Changed|
|Checkbox.Circular Disabled.normal.chromium.png|4711| Changed|
|Checkbox.Disabled.normal.chromium.png|4705| Changed|
|Checkbox.Indeterminate - Dark Mode.normal.chromium.png|877| Changed|
|Checkbox.Indeterminate.normal.chromium.png|4776| Changed|
|Checkbox.Indeterminate.indeterminate.chromium.png|4847| Changed|
|Checkbox.Default - Dark Mode.normal.chromium.png|814| Changed|
|Checkbox.Default - RTL.normal.chromium.png|4751| Changed|
|Checkbox.Circular Disabled - Dark Mode.normal.chromium.png|760| Changed|
|Checkbox.Circular Disabled Checked - Dark Mode.normal.chromium.png|799| Changed|
|Checkbox.Disabled - Dark Mode.normal.chromium.png|739| Changed|
|Checkbox.Disabled Checked.normal.chromium.png|4721| Changed|

</details>

  <details><summary><b>Divider</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Divider.With Svg - RTL.normal.chromium.png|3| Changed|
|Divider.With Svg.normal.chromium.png|3| Changed|
|Divider.Vertical With Svg.normal.chromium.png|4| Changed|

</details>

  <details><summary><b>Label</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Label.Required - RTL.normal.chromium.png|11| Changed|
|Label.Size Small.normal.chromium.png|661| Changed|

</details>

  <details><summary><b>Link</b> 10 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Link.Default.default.chromium.png|0| Added|
|Link.Default.hover.chromium.png|0| Added|
|Link.Default.pressed.chromium.png|0| Added|
|Link.Inline.default.chromium.png|0| Added|
|Link.Inline.hover.chromium.png|0| Added|
|Link.Inline.pressed.chromium.png|0| Added|
|Link.Subtle.default.chromium.png|0| Added|
|Link.Subtle.hover.chromium.png|0| Added|
|Link.Subtle.pressed.chromium.png|0| Added|
|Link.With Long Text.default.chromium.png|0| Added|

</details>

  <details><summary><b>MenuList</b> 35 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|MenuList.Checkbox - Dark Mode.1st selected.chromium.png|3| Changed|
|MenuList.Checkbox.2nd selected.chromium.png|3| Changed|
|MenuList.Checkbox - RTL.2nd selected.chromium.png|3| Changed|
|MenuList.Checkbox With Icons - RTL.1st selected.chromium.png|6| Changed|
|MenuList.Checkbox With Icons - RTL.normal.chromium.png|4| Changed|
|MenuList.Checkbox.1st selected.chromium.png|2| Changed|
|MenuList.Radio - RTL.2nd selected.chromium.png|2| Changed|
|MenuList.Checkbox - Dark Mode.2nd selected.chromium.png|3| Changed|
|MenuList.Radio With Icons - RTL.2nd selected.chromium.png|17| Changed|
|MenuList.Radio With Icons.1st selected.chromium.png|5| Changed|
|MenuList.Radio - RTL.1st selected.chromium.png|2| Changed|
|MenuList.Checkbox With Icons.normal.chromium.png|4| Changed|
|MenuList.Radio.2nd selected.chromium.png|2| Changed|
|MenuList.Radio - Dark Mode.2nd selected.chromium.png|3| Changed|
|MenuList.Checkbox - RTL.1st selected.chromium.png|2| Changed|
|MenuList.Checkbox With Icons.1st selected.chromium.png|5| Changed|
|MenuList.Checkbox With Icons - RTL.2nd selected.chromium.png|17| Changed|
|MenuList.Radio.1st selected.chromium.png|2| Changed|
|MenuList.Radio With Icons - RTL.1st selected.chromium.png|6| Changed|
|MenuList.Checkbox With Icons.2nd selected.chromium.png|17| Changed|
|MenuList.With Submenu - Dark Mode.2nd selected.chromium.png|23| Changed|
|MenuList.With Icons.normal.chromium.png|4| Changed|
|MenuList.Radio With Icons.2nd selected.chromium.png|17| Changed|
|MenuList.Radio - Dark Mode.1st selected.chromium.png|3| Changed|
|MenuList.With Submenu - Dark Mode.1st selected.chromium.png|23| Changed|
|MenuList.Radio With Icons.normal.chromium.png|4| Changed|
|MenuList.With Submenu.1st selected.chromium.png|34| Changed|
|MenuList.Radio With Icons - RTL.normal.chromium.png|4| Changed|
|MenuList.With Submenu.normal.chromium.png|23| Changed|
|MenuList.With Icons.hover menuitem.chromium.png|3| Changed|
|MenuList.With Submenu - RTL.normal.chromium.png|23| Changed|
|MenuList.With Submenu - Dark Mode.normal.chromium.png|21| Changed|
|MenuList.With Submenu - RTL.1st selected.chromium.png|706| Changed|
|MenuList.With Submenu.2nd selected.chromium.png|32| Changed|
|MenuList.With Submenu - RTL.2nd selected.chromium.png|949| Changed|

</details>

  <details><summary><b>Slider</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Slider.Size Small.rightArrow.chromium.png|1| Changed|
|Slider.Size Small.normal.chromium.png|1| Changed|

</details>

  <details><summary><b>Switch</b> 34 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Switch.Checked.hover.chromium.png|5150| Changed|
|Switch.Default - RTL.hover.chromium.png|5044| Changed|
|Switch.Default - RTL.normal.chromium.png|5009| Changed|
|Switch.Default.hover.chromium.png|5044| Changed|
|Switch.Default.normal.chromium.png|5009| Changed|
|Switch.Disabled - Dark Mode.normal.chromium.png|909| Changed|
|Switch.Checked - Dark Mode.normal.chromium.png|1130| Changed|
|Switch.Checked - Dark Mode.hover.chromium.png|1169| Changed|
|Switch.Disabled Checked.hover.chromium.png|5047| Changed|
|Switch.Disabled Checked.normal.chromium.png|5047| Changed|
|Switch.Disabled.hover.chromium.png|4914| Changed|
|Switch.Disabled - RTL.hover.chromium.png|4914| Changed|
|Switch.Checked - RTL.normal.chromium.png|5132| Changed|
|Switch.Disabled - RTL.normal.chromium.png|4914| Changed|
|Switch.Checked - RTL.hover.chromium.png|5150| Changed|
|Switch.Disabled Checked - Dark Mode.normal.chromium.png|1074| Changed|
|Switch.Default - Dark Mode.hover.chromium.png|1071| Changed|
|Switch.Default - Dark Mode.normal.chromium.png|1019| Changed|
|Switch.Disabled.normal.chromium.png|4914| Changed|
|Switch.Checked.normal.chromium.png|5132| Changed|
|Switch.Disabled Checked - Dark Mode.hover.chromium.png|1074| Changed|
|Switch.Disabled - Dark Mode.hover.chromium.png|909| Changed|
|Switch.Label Above - RTL.hover.chromium.png|0| Removed|
|Switch.Label Above - RTL.normal.chromium.png|0| Removed|
|Switch.Label Above.hover.chromium.png|0| Removed|
|Switch.Label Above.normal.chromium.png|0| Removed|
|Switch.Label After - RTL.hover.chromium.png|0| Removed|
|Switch.Label After - RTL.normal.chromium.png|0| Removed|
|Switch.Label After.hover.chromium.png|0| Removed|
|Switch.Label After.normal.chromium.png|0| Removed|
|Switch.Label Before - RTL.hover.chromium.png|0| Removed|
|Switch.Label Before - RTL.normal.chromium.png|0| Removed|
|Switch.Label Before.hover.chromium.png|0| Removed|
|Switch.Label Before.normal.chromium.png|0| Removed|

</details>

  <details><summary><b>Text</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Text.Block.normal.chromium.png|958| Changed|

</details>

  <details><summary><b>TextInput</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|TextInput.Content Start End - RTL.normal.chromium.png|119| Changed|
|TextInput.Content Start End.normal.chromium.png|116| Changed|
|TextInput.Size Large.normal.chromium.png|3| Changed|
|TextInput.Content Start End - Dark Mode.normal.chromium.png|102| Changed|

</details>

<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

> ## Basic Recomposition

Reviewer: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31861&commitId=4ecb798a0d12e7092808dbd7f932d29a50a5993c&env=prod&clientType=FLUENTUI)

  <details><summary><b>Pivot - Overflow</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Pivot - Overflow.Root.Narrow - Last tab selected.chromium.png|0| Added|
|Pivot - Overflow.Root.Narrow - Overflow menu.chromium.png|0| Added|
|Pivot - Overflow.Tabs - RTL.Narrow - Last tab selected.chromium.png|0| Added|
|Pivot - Overflow.Tabs - RTL.Narrow - Overflow menu.chromium.png|0| Added|

</details>

<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

> ## Basic Recomposition

Reviewer: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31861&commitId=4ecb798a0d12e7092808dbd7f932d29a50a5993c&env=prod&clientType=FLUENTUI)

  <details><summary><b>Avatar Converged</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|5| Changed|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|6| Changed|
|Avatar Converged.basic - High Contrast.normal.chromium.png|0| Added|
|Avatar Converged.basic - RTL.normal.chromium.png|0| Removed|

</details>

  <details><summary><b>SwatchPicker Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.Spacing.default.chromium.png|0| Removed|
|SwatchPicker Converged.spacing.default.chromium.png|0| Removed|

</details>

<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

> Fluent UI React V9 leans into Recomposition as one of its primary methods of customization. That is, reassemble parts of components to inherit most behaviors but also change some. We wanted to change 2 things for the Nav Divider component to help consumers quickly and easily implement specified behaviors:

Reviewer 2: ```suggestion
Fluent UI React V9 leans into recomposition as one of its primary methods of customization. That is, reassemble parts of components to inherit most behaviors but also change some. We wanted to change 2 things for the Nav Divider component to help consumers quickly and easily implement specified behaviors:
```

Author: fixed🙂

> Fluent UI React V9 leans into Recomposition as one of its primary methods of customization. That is, reassemble parts of components to inherit most behaviors but also change some. We wanted to change 2 things for the Nav Divider component to help consumers quickly and easily implement specified behaviors:

Reviewer 3: I would say that the `Nav` is an example of what is being talked about above.

```suggestion
Fluent UI React V9 leans into recomposition as one of its primary methods of customization. That is, reassemble parts of components to inherit most behaviors but also change some. Below, we will go through an example where we needed to do this in the library with the `NavDivider` component.

We wanted to change 2 things for the `NavDivider` component, when compare to the regular `Divider`, to help consumers quickly and easily implement specified behaviors:
```

Author: Changed the sentence structure slightly to prefer an active voice and shortened 🙂

1.  Change the default appearance prop to 'strong'

>     - The default Divider appearance happens to be the same color as the NavDrawer

Reviewer 3: ```suggestion
    - The default `Divider` appearance happens to be the same color as the `NavDrawer`
```

Author: added 🙂

2.  Change some default styles

>     - FlexGrow needs to be changed from 1 to 0

Reviewer 3: ```suggestion
    - `flexGrow` needs to be changed from 1 to 0
```

Author: added 🙂

    - Add 4 pixels of top and bottom margin

Assuming you've [scaffolded out the component](https://github.com/microsoft/fluentui/blob/8a3aa5f6200012d58ed80a833d8690d77935a48b/docs/react-v9/contributing/command-cheat-sheet.md?plain=1#L8), the best place to start is in the type definition.

> We want to inherit as much as we can from Divider, it simply becomes:

Reviewer 3: ```suggestion
We want to inherit as much as we can from `Divider`, it simply becomes:
```

Author: updated 🙂

> ```

Reviewer 3: ```suggestion
```ts
```

Author: TIL 🙏

// NavDividerTypes.ts
import type { DividerProps, DividerState } from '@fluentui/react-divider';

// Remove the NavDivider slot type definition

/**
 * NavDivider Props
 */
export type NavDividerProps = DividerProps;

/**
 * State used in rendering NavDivider
 */
export type NavDividerState = DividerState;
```

> We can also delete the `renderNavDivider.ts` file all together and update our wrapper component to call the base `renderDivider` function.

Reviewer 4: Great doc! I think some context to let users know that the render func can be overridden if wanted, but it wasn't required here - some examples might be if we need to include our own context wrappers for the extended component and adding additional UI elements/slots or logic.

Author: Yep - I forsee an "Advanced Recomposition" document coming soon :-)

> ```

Reviewer 3: ```suggestion
```tsx
```

// NavDivider.tsx
import * as React from 'react';
import { useNavDivider_unstable } from './useNavDivider';
import { useNavDividerStyles_unstable } from './useNavDividerStyles.styles';
import { renderDivider_unstable } from '@fluentui/react-divider';

import type { ForwardRefComponent } from '@fluentui/react-utilities';
import type { NavDividerProps } from './NavDivider.types';

/**
 * NavDivider component
 */
export const NavDivider: ForwardRefComponent<NavDividerProps> = React.forwardRef((props, ref) => {
  const state = useNavDivider_unstable(props, ref);

  useNavDividerStyles_unstable(state);

  return renderDivider_unstable(state); // Rather than render*Nav*Divider
});

NavDivider.displayName = 'NavDivider';
```

Now that the scaffolding has been updated, lets assign our new prop opinions in the use hook. We can remove most of the boiler plate code because we're just going to call the base `useDivider` hook:

> ```

Reviewer 3: ```suggestion
```ts
```

// useNavDivider.ts
import * as React from 'react';
import type { NavDividerProps, NavDividerState } from './NavDivider.types';
import { useDivider_unstable } from '@fluentui/react-divider';

/**
 * Create the state required to render NavDivider.
 *
 * The returned state can be modified with hooks such as useNavDividerStyles_unstable,
 * before being passed to renderNavDivider_unstable.
 *
 * @param props - props from this instance of NavDivider
 * @param ref - reference to root HTMLDivElement of NavDivider
 */
export const useNavDivider_unstable = (props: NavDividerProps, ref: React.Ref<HTMLElement>): NavDividerState => {
  return useDivider_unstable({ appearance: 'strong', ...props }, ref);
};
```

> The last task is to add our new styling opinions. Pay special note to the location of `useDividerStyles`. We need to generate the base styles, but must add them _after_ merging the Nav specific styles such that the Nav styling opinions can 'win' and don't get over written by the base styles in the event of a collision (as there is with `flexGrow`.

Reviewer 5: ```suggestion
The last task is to add our new styling opinions. Pay special note to the location of `useDividerStyles`. We need to generate the base styles, but must add them _after_ merging the Nav specific styles such that the Nav styling opinions can 'win' and don't get overwritten by the base styles in the event of a collision (as there is with `flexGrow`).
```

Author: removed the space 🙂

> The last task is to add our new styling opinions. Pay special note to the location of `useDividerStyles`. We need to generate the base styles, but must add them _after_ merging the Nav specific styles such that the Nav styling opinions can 'win' and don't get over written by the base styles in the event of a collision (as there is with `flexGrow`.

Reviewer 3: ```suggestion
The last task is to add our new styling opinions. Pay special note to the location of `useDividerStyles`. We need to generate the base styles, but must add them _after_ merging the `Nav` specific styles such that the `Nav` styling opinions can 'win' and don't get overwritten by the base styles in the event of a collision (as there is with `flexGrow`).
```

Author: fixed 🙂

> ```

Reviewer 3: ```suggestion
```ts
```
